gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java

```diff
@@ -187,9 +194,59 @@ public Optional<Dag<JobExecutionPlan>> validateAndHandleConcurrentExecution(Conf
    * @param allowConcurrentExecution
    * @return true if the {@link FlowSpec} allows concurrent executions or if no other instance of the flow is currently RUNNING.
    */
-  private boolean isExecutionPermitted(FlowStatusGenerator flowStatusGenerator, String flowGroup, String flowName,
-      boolean allowConcurrentExecution, long flowExecutionId) {
-    return allowConcurrentExecution || !flowStatusGenerator.isFlowRunning(flowName, flowGroup, flowExecutionId);
+  private boolean isExecutionPermitted(String flowGroup, String flowName, boolean allowConcurrentExecution)
+      throws IOException {
+    return allowConcurrentExecution || !isFlowRunning(flowGroup, flowName, dagManagementStateStore);
+  }
+
+  /**
+   * Returns true if any previous execution for the flow determined by the provided flowGroup, flowName is running.
+   * We ignore the execution that has the provided flowExecutionId. We also ignore the flows that are running beyond
+   * the job start deadline and flow finish deadline.
+   */
+  @VisibleForTesting
+  static boolean isFlowRunning(String flowGroup, String flowName, DagManagementStateStore dagManagementStateStore)
+      throws IOException {
+    List<FlowStatus> flowStatusList = dagManagementStateStore.getAllFlowStatusesForFlow(flowGroup, flowName);
+
+    if (flowStatusList == null || flowStatusList.isEmpty()) {
+      return false;
+    }
+
+    for (FlowStatus flowStatus : flowStatusList) {
+      ExecutionStatus flowExecutionStatus = flowStatus.getFlowExecutionStatus();
+      log.info("Verifying if {} is running...", flowStatus);
+
+      if (flowExecutionStatus == COMPILED || flowExecutionStatus == PENDING
+          || flowExecutionStatus == PENDING_RESUME || flowExecutionStatus == RUNNING) {
```

Comment:
let's code more defensively by confirm specific statues we expect and flagging those that are unrecognized.  e.g. what about ORCHESTRATED?  -
```
if (fES == COMPLETED || fES == FAILED) {
  // no-op
} else if (fES == COMPILED || PENDING_RESUME || RUNNING) {
  ...
} else {
  log.error("unknown fES...");
}
```